### PR TITLE
docs: LIMIT n BY modifier

### PR DESCRIPTION
**Pull request by f1yegor**
Relates to https://github.com/yandex/ClickHouse/pull/293

### CODE REVIEW

**Comment by robot-metrika-test** [comment]
Can one of the admins verify this patch?

File: doc/reference_en.html [hunk 1]
```diff
@@ -2022,6 +2022,25 @@ <h3 class="not-for-contents">1. Why not to use systems like map-reduce?</h3>
 If you have an ORDER BY clause with some small LIMIT after a GROUP BY, then ORDER BY will not consume significant amount of memory. But if no LIMIT is provided, don't forget to turn on external sorting (%%max_bytes_before_external_sort%%).
 
 
+<h5>LIMIT N BY modifier</h5>
+
+LIMIT %%N%% BY %%COLUMN%% allows you to restrict top %%N%% rows per each group of %%COLUMN%%
```
Comment (alexey-milovidov) [inline doc/reference_en.html]: No mention about that key for LIMIT BY could contain arbitary number of columns and expressions.

File: doc/reference_en.html [hunk 2]
```diff
@@ -2022,6 +2022,25 @@ <h3 class="not-for-contents">1. Why not to use systems like map-reduce?</h3>
...
+Example:
+
+%%SELECT
+    domainWithoutWWW(URL) AS domain,
+    domainWithoutWWW(REFERRER_URL) AS referrer,
+    count() cnt
+FROM hits
+GROUP BY domain, referrer
+ORDER BY cnt DESC
+LIMIT 5 BY domain
```
Comment (alexey-milovidov) [inline doc/reference_en.html]: No mention about LIMIT BY is unrelated to LIMIT and query could even contain both clauses (ordinary LIMIT could go after LIMIT BY).

File: doc/reference_en.html [hunk 3]
```diff
@@ -2022,6 +2022,25 @@ <h3 class="not-for-contents">1. Why not to use systems like map-reduce?</h3>
...
+%%
+
+will select top 5 referrers for each domain.
+
+
```
Comment (alexey-milovidov) [inline doc/reference_en.html]: What about adding this to russian doc too?

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]

**Comment by alexey-milovidov** [comment]
Thanks! Almost Ok. Just few little changes to do.

File: doc/reference_en.html (hunk 3 above)
Comment (f1yegor) [inline doc/reference_en.html]: Will add after approve of english version.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
```
%%LIMIT N BY COLUMNS%% is unrelated to %%LIMIT BY%% clause.
```

- typo.

**Pull request merged**